### PR TITLE
API version 2.5 changes

### PR DESCRIPTION
**Pull request by bhelx**
These are all the changes needed to bump us to API version 2.5. Mostly some added fields and we added the `forceCollect` endpoint.

### CODE REVIEW

File: src/main/java/com/ning/billing/recurly/model/BillingInfo.java [hunk 1]
```diff
@@ -398,7 +424,8 @@ public int hashCode() {
                 year,
                 month,
                 firstSix,
-                lastFour
+                lastFour,
+                updatedAt
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/BillingInfo.java]: Missing `geoCode`.

Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/BillingInfo.java]: Missing `geoCode`.

File: src/main/java/com/ning/billing/recurly/model/Plan.java [hunk 2]
```diff
@@ -202,6 +205,14 @@ public void setTrialIntervalUnit(final Object trialIntervalUnit) {
         this.trialIntervalUnit = stringOrNull(trialIntervalUnit);
     }
 
+    public Boolean getTrialRequiresBillingInfo() {
+        return this.trialRequiresBillingInfo;
+    }
+
+    public void setTrialRequiresBillingInfo(final Object trialRequiresBillingInfo) {
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/Plan.java]: Missing `toString`/`equals`/`hashCode`.

File: src/main/java/com/ning/billing/recurly/model/Transaction.java [hunk 3]
```diff
@@ -188,11 +256,17 @@ public String toString() {
...
         sb.append(", createdAt=").append(createdAt);
         sb.append(", details=").append(details);
         sb.append(", paymentMethod=").append(paymentMethod);
         sb.append(", collectedAt=").append(collectedAt);
         sb.append(", updatedAt=").append(updatedAt);
+        sb.append(", origin=").append(origin);
+        sb.append(", gatewayType=").append(gatewayType);
+        sb.append(", message=").append(message);
+        sb.append(", failureType=").append(failureType);
+        sb.append(", approvalCode=").append(approvalCode);
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/Transaction.java]: Missing `gatewayErrorCodes`.

File: src/main/java/com/ning/billing/recurly/model/Transaction.java [hunk 4]
```diff
@@ -243,6 +320,21 @@ public boolean equals(final Object o) {
...
+        if (approvalCode != null ? !approvalCode.equals(that.approvalCode) : that.approvalCode != null) {
+            return false;
+        }
+        if (failureType != null ? !failureType.equals(that.failureType) : that.failureType != null) {
+            return false;
+        }
+        if (gatewayType != null ? !gatewayType.equals(that.gatewayType) : that.gatewayType != null) {
+            return false;
+        }
+        if (gatewayErrorCodes != null ? !gatewayErrorCodes.equals(that.gatewayErrorCodes) : that.gatewayErrorCodes != null) {
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/Transaction.java]: Missing `message`.

File: src/main/java/com/ning/billing/recurly/model/Transaction.java [hunk 5]
```diff
@@ -257,6 +349,7 @@ public int hashCode() {
                 taxInCents,
                 currency,
                 description,
+                productCode,
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/Transaction.java]: Missing some elements 😏

**Comment by bhelx** [comment]
Thanks for the quick response @pierre! Just published an update.

**Comment by pierre** [comment]
Was this the last changes needed for 2.5 (Travis hasn't recovered yet)?

**Comment by bhelx** [comment]
@pierre yes this is all of it. I will check to see what is going on with travis. I think one test is being finicky.

**Comment by bhelx** [comment]
@pierre spoke too soon. Removing two fields in Transaction: https://github.com/killbilling/recurly-java-library/pull/182

**Pull request merged**